### ISSUE TITLE
[BUG] I think the price can go up to 660 only and it is showing 661 in the chart

### ISSUE THREAD

**Issue by KennethKinLum** [elxris/Turnip-Calculator#134]
### Version of the tool
v2.1.2-build.14

### Describe the bug
I ran the code at https://gist.github.com/Treeki/85be14d297c80c8b3c0a76375743325b
with `./a.out 2 <seed>` with the seed 0 to 1,300,000
and never was able to get 661.  660 showed up many times.  I will post the results in the next comment.

Some posts on the Internet said it is only up to 6x the price can be, and with the purchase price of the island master's purchase price of turnips as a base?  So if it is 90, then 90 x 6 = 540 max, and if the purchase price is 110, then 110 x 6 = 660 max.

**Comment by KennethKinLum** [comment]
the first number is seed, and the next set of data is the result running with the seed:

```
326
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   96  137  660  152   65   82
      91  200  180   91   63   80
9772
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   99  213  177   58   68   62
     141  660  145   57   71   73
38838
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   99   90  133  660  137   72
      94   86  205  200   72   61
48417
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   98   88   80  115  660  151
      93   83   75  181  191   71
53083
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   97   89  190  159   66   63
      94  116  660  112   95   87
76337
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   99   90  100  660  119   90
      95   86  196  166   47   96
78834
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   99   90   83  197  212   64
      94   87  147  660  129   92
84720
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   95  105  660  128   56   54
      92  216  183   46   58   86
94725
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   97   88  202  188   68   57
      94  148  660  147   85   63
106227
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   99   91   82   73  192  185
      95   87   78  126  660  113
116355
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   99   90   83  111  660  140
      95   86   79  163  206   98
119160
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   99   91   80  142  660  107
      96   86   76  210  200   61
136238
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   96   89   79   70  216  215
      92   85   74  150  660  108
150230
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   99   91  182  191   49   85
      95  110  660  144   90   57
178084
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   98  202  220   58   47   54
     122  660  151   53   70   67
219152
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   96   87   77  170  161   46
      91   82  138  660  109   57
230113
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   94   86   78   70  218  187
      90   81   73  141  660  135
276863
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   95   86  110  660  111   75
      90   82  155  188   88   71
291879
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   99  181  186   50   75   68
     106  660  120   86   60   55
319404
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   96  158  218   55   64   77
     131  660  111   88   71   56
322452
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   96  198  197   77   76   75
     132  660  122   74   74   60
325330
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   94   85   76  119  660  110
      91   80   71  194  194   75
339841
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   97  180  188   59   93   59
     137  660  113   80   95   45
371853
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   95   87  123  660  113   87
      91   82  211  167   54   73
432408
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   96   88   80  104  660  131
      92   83   76  196  192   60
558655
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   96  163  169   73   69   57
     104  660  128   48   83   72
658276
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   97  210  219   45   57   46
     137  660  137   71   68   94
663748
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   99   91   82  105  660  130
      95   86   78  161  204   70
682090
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   96  191  178   88   64   70
     142  660  122   74   52   48
706855
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   97   88  102  660  139   62
      92   83  172  209   88   57
728966
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   94   87   79   71  203  168
      91   83   75  121  660  111
743733
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   99   91   83  138  660  117
      95   88   78  156  184   75
758237
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   99   90  180  220   97   48
      94  149  660  141   98   97
777818
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   95   86   78  184  175   80
      90   83  107  660  131   67
813788
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   97  174  169   73   83   64
     140  660  154   99   96   52
819942
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   97   89  123  660  126   69
      93   85  218  196   58   83
825763
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   97   88   77  105  660  131
      92   83   74  167  185   94
826017
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   99   88   81  101  660  150
      94   85   77  159  193   63
845772
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   95   85  172  209   68   82
      90  118  660  115   81   68
866532
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   99  204  216   73   97   85
     134  660  111   93   53   57
879360
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   96   86   78  121  660  123
      91   82   74  183  214   70
902263
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   95  197  205   54   49   46
     140  660  119   65   78   57
969062
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   97  107  660  106   73   60
      92  194  207   59   95   60
981132
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   95  166  220   77   74   56
     152  660  108   93   52   71
995099
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   96   88   77  170  160   61
      91   83  105  660  104   91
1000449
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   95   87   79  200  157   79
      91   82  144  660  147   45
1003757
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   97  119  660  142   53   97
      92  203  179   45   68   73
1029958
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   97   89   79  140  660  139
      93   83   75  200  157   92
1107343
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   99   92   82  172  168   76
      95   87  105  660  107   84
1114673
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   94  187  189   98   79   92
     106  660  119   89   75   75
1154146
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   99   89  138  660  147   71
      95   85  170  167   53   69
1171479
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   98   89  111  660  111   56
      94   84  191  214   73   59
1181944
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   97   86   76   66  197  165
      92   81   71  122  660  150
1183467
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   96  205  207   66   61   68
     100  660  142   87   96   90
1185335
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   99  196  190   63   70   97
     120  660  137   53   85   72
1257420
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   98   91   81  135  660  132
      94   86   77  200  157   64
1264666
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   94  114  660  109   82   58
      90  158  157   87   57   61
1267859
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   97   88   80  160  178   82
      92   83  112  660  150   64
1271250
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   99  180  176   73   50   88
     143  660  107   95   79   87
1273934
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   99   91   81  144  660  141
      95   86   76  176  204   77
1326333
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   98   90   80  143  660  150
      94   85   75  194  162   55
1329274
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   99   89  220  176   73   94
      94  102  660  134   97   61
1338451
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   98   90   83   74  215  205
      94   86   78  131  660  110
1343496
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   96   87  120  660  136   94
      91   83  191  169   53   48
1359315
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   96   89  208  217   95   83
      92  123  660  102   61   92
1382333
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   96   88   78  123  660  151
      92   84   73  206  184   81
1391004
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   98   91  151  660  152   56
      95   86  173  161   84   90
1396699
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   96   87   78   69  202  158
      92   82   74  123  660  111
1398979
Pattern 1:
Sun  Mon  Tue  Wed  Thu  Fri  Sat
110   96  138  660  117   53   99
      91  159  191   59   74   63
```

**Comment by elxris** [comment]
Thanks a lot for opening this, I didn't notice before, but you're completely right!

I was able to find the precision issue that was affecting the calculations. It's something deeply rooted in how the number generator works, but I was able to find a solution that basically emulates the number as it was generated by the original source. 👍 

I'm happy with the result. I can't thank you enough.

![image](https://user-images.githubusercontent.com/1855014/83980028-02b46300-a8d8-11ea-86a7-520a3783501e.png)
